### PR TITLE
MGMT-3539: get k8s cluster object by name and namespace

### PR DESCRIPTION
**Pull request by RazRegev**

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -618,6 +628,8 @@ func (b *bareMetalInventory) RegisterClusterInternal(ctx context.Context, params
 		AdditionalNtpSource:      swag.StringValue(params.NewClusterParams.AdditionalNtpSource),
 		Operators:                params.NewClusterParams.Operators,
 		HighAvailabilityMode:     params.NewClusterParams.HighAvailabilityMode,
+		KubeAPIName:              &kubeAPIKey.Name,
```
Comment (filanov) [inline internal/bminventory/inventory.go:631]: The name is confusing, why not just `KubeName`, `KubeNamespace`?

File: internal/cluster/cluster.go [hunk 2]
```diff
@@ -93,6 +97,7 @@ type API interface {
 	DeleteClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error
 	PermanentClustersDeletion(ctx context.Context, olderThen strfmt.DateTime, objectHandler s3wrapper.API) error
 	UpdateInstallProgress(ctx context.Context, c *common.Cluster, progress string) *common.ApiErrorResponse
+	GetK8SClusterObject(ctx context.Context, c *common.Cluster) (*v1alpha1.Cluster, error)
```
Comment (filanov) [inline internal/cluster/cluster.go:100]: Where do you think it's going to be used?

File: swagger.yaml [hunk 3]
```diff
@@ -3588,6 +3588,14 @@ definitions:
         $ref: '#/definitions/operators'
         x-go-custom-tag: gorm:"embedded;embedded_prefix:operators_"
         x-nullable: true
+      kubeAPIName:
```
Comment (filanov) [inline swagger.yaml:3591]: it should be internal data under https://github.com/openshift/assisted-service/blob/master/internal/common/db.go

**Comment by filanov** [comment]
The task talked about creating a cluster and storing kube key internally.

**Comment by RazRegev** [comment]
will open another pr

**Pull request closed without merging**